### PR TITLE
Convert `viewport` to TypeScript

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Convert `viewport` to TypeScript

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [ ] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+};
+
+const onMessage = (
+	respond: RespondProxy,
+	start: unknown,
+	iframe: HTMLIFrameElement | undefined,
+): void => {
+	if (!iframe) return;
+	if (typeof start !== 'boolean') return;
```
[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/messenger/viewport.ts:89]: One line returns don't bother me too much, but I understand there's room to misinterpret so I'll update it!

ps we do have this related rule however:
https://github.com/guardian/configs/pull/135

File: static/src/javascripts/projects/commercial/modules/messenger.ts [hunk 2]
```diff
@@ -98,16 +124,39 @@ export type RegisterListener = (
 	},
 ) => void;
 
+type ListenerOptions = {
+	window?: WindowProxy;
+	persist?: boolean;
```
[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/messenger.ts]: Thanks good shout! Will remove

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/messenger/viewport.ts]: > ps we do have this related rule however: [guardian/configs#135](https://github.com/guardian/configs/pull/135)

Ah nice!

**Review by chrislomaxjones — APPROVED** [review]
Great improvement!

**Comment by arelra** [comment]
> Great improvement!

Thanks for your help! This is quite the claim - but I think we understand messenger now 😄 😱

**Pull request merged**